File: poverty/poverty.rb
class County
  attr_accessor :name, :percent, :num, :median_income

  def initialize(name, percent, num, median_income)
    @name = name
    @percent = percent.to_f
    @num = num.to_i
    @median_income = median_income.to_i
  end

  def self.all_counties
    file_WA = File.open('est14_WA.txt')
    data_array = []
    file_WA.each do |line|
      data_array.push(County.new(line[193..217], line[76..79], line[49..56], line[133..138]))
    end
    return data_array
  end

  def self.find_highest_pov
    highest = 0
    county_name = ""
    count = 0
    income = 0
    all_counties.each do |county|
      if county.percent > highest
        highest = county.percent
        county_name = county.name
        count = county.num
        income = county.median_income
      end
    end
    return "#{highest}% poverty in #{county_name}. Count is #{count} and median income is $#{income}"
  end
  puts self.find_highest_pov

  def self.find_lowest_pov
    lowest = 100
    county_name = ""
    count = 0
    income = 0
    all_counties.each do |county|
      if county.percent < lowest
        lowest = county.percent
        county_name = county.name
        count = county.num
        income = county.median_income
      end
    end
    return "#{lowest}% poverty in #{county_name}. Count is #{count} and median income is $#{income}"
  end
  puts self.find_lowest_pov
end

File: poverty/poverty.py
def highest_state_poverty(data):
    highest_name = ""
    highest_percent = 0
    highest_num = 0
    highest_median_income = 0
    file = open(data, 'r')
    for line in file:
        name = line[193:218].strip()
        percent = float(line[76:80])
        num = line[49:57].strip()
        median_income = line[133:139].strip()
        if percent > highest_percent:
            highest_percent = percent
            highest_name = name
            highest_num = num
            highest_median_income = median_income
    print "The county with the highest child poverty is %s, with %s percent. That amounts to %s children. The median_income for the county is $%s" % (highest_name, highest_percent, highest_num, highest_median_income)

def lowest_state_poverty(data):
    lowest_name = ""
    lowest_percent = 100
    lowest_num = 0
    lowest_median_income = 0
    file = open(data, 'r')
    for line in file:
        name = line[193:218].strip()
        percent = float(line[76:80])
        num = line[49:57].strip()
        median_income = line[133:139].strip()
        if percent < lowest_percent:
            lowest_percent = percent
            lowest_name = name
            lowest_num = num
            lowest_median_income = median_income
    print "The county with the lowest child poverty is %s, with %s percent. That amounts to %s children. The median_income for the county is $%s" % (lowest_name, lowest_percent, lowest_num, lowest_median_income)

print lowest_state_poverty('est14_GA.txt')
print highest_state_poverty('est14_GA.txt')
